app/services/sso/openid_connect.rb

```diff
@@ -4,27 +4,48 @@ class OpenidConnect < Base
     attr_reader :current_user
 
     def available?
+      if session[:sso_method] == "SSO::OpenidConnect"
+        session.delete(:sso_method) unless session[:user]
+        return true
+      end
       ((controller.api_request? && bearer_token_set?) || http_token.present?) && valid_issuer?
     end
 
     def authenticate!
+      self.has_rendered = true
+      controller.redirect_to login_url
+    end
+
+    def authenticated?
       payload = jwt_token.decode
-      return nil if payload.nil?
+      return false if payload.nil?
       user = find_or_create_user_from_jwt(payload)
       @current_user = user
```

Comment:
is the `@current_user` used anywhere? I couldn't find any such case

Reply:
The `@current user` is used here: https://github.com/theforeman/foreman/blob/develop/app/controllers/concerns/foreman/controller/authentication.rb#L56 and from the way I had interpreted it was to have declared an `attr_reader` for it. So, when it is called it directly gets the value.